lib/ToolbarBadgeHub.jsm

```diff
@@ -70,12 +83,16 @@ class _ToolbarBadgeHub {
     this._blockMessageById = blockMessageById;
     this._addImpression = addImpression;
     this._dispatch = dispatch;
-    this.state = {};
     // Need to wait for ASRouter to initialize before trying to fetch messages
     await waitForInitialized;
     this.messageRequest("toolbarBadgeUpdate");
     // Listen for pref changes that could trigger new badges
     Services.prefs.addObserver(this.prefs.WHATSNEW_TOOLBAR_PANEL, this);
+    const _intervalId = setInterval(
+      () => this.messageRequest("toolbarBadgeUpdate"),
```

Comment:
You could add some logic here to clear / set the pref

Reply:
See https://docs.google.com/document/d/1r_Zx75budKb4ZgBKEyIuXSR47vm-OkW0FE50PwS53Q0/edit#heading=h.28r6948w8f9h re:our discussion